### PR TITLE
Add a simple property-based test using Hypothesis

### PR DESCRIPTION
**Pull request by Zac-HD**
This is a trivial property-based test, meant to get us started on #2444 by checking that installation and CI is working.

### CODE REVIEW

**Comment by Zac-HD** [comment]
And... [the test fails](https://github.com/napari/napari/pull/2469/checks?check_run_id=2190305569#step:7:1850-1871), because `guess_rgb()` *does not* check if the last dimension is of size 3 or 4 (as documented); it just checks if it's less than five.  Multi-dimensional non-RGB arrays with a narrow trailing dimension are therefore mishandled:

```python-traceback
  =================================== FAILURES ===================================
  ___________________________ test_guess_rgb_property ____________________________
  
      @given(shape=array_shapes(min_dims=3, min_side=0))
      def test_guess_rgb_property(shape):
  >       assert guess_rgb(shape) == (shape[-1] in (3, 4))
  E       assert True == False
  E         +True
  E         -False
  
  napari/layers/image/_tests/test_image_utils.py:31: AssertionError
  ---------------------------------- Hypothesis ----------------------------------
  Falsifying example: test_guess_rgb_property(
      shape=(0, 0, 0),
  )
```

(note that Hypothesis has reported a *minimal* counterexample :grin: )

https://github.com/napari/napari/blob/102a7e8f845893c874d2b86f9371d41130100b89/napari/layers/image/_image_utils.py#L6-L24

File: docs/developers/testing.md [hunk 1]
```diff
@@ -90,6 +90,21 @@ Writing tests for new code is a critical part of keeping napari maintainable as
 it grows. Tests are written in files whose names
 begin with `test_*` and which are contained in one of the `_tests` directories.
 
+### Property-based testing with Hypothesis
+
+Property-based tests allow you to test that "for any X, ..." - with a much nicer
+developer experience than using truly random data.  We use Hypothesis for unit or
+integration tests where there are simple properties like "x == load(save(x))` or
```
Comment (goanpeca) [inline docs/developers/testing.md]: typo?

```
"x == load(save(x))`
```
-> 
```
`x == load(save(x))`
```

File: napari/layers/image/_image_utils.py [hunk 2]
```diff
@@ -21,7 +21,7 @@ def guess_rgb(shape):
     ndim = len(shape)
     last_dim = shape[-1]
 
-    return ndim > 2 and last_dim < 5
+    return ndim > 2 and last_dim in (3, 4)
```
Comment (goanpeca) [inline napari/layers/image/_image_utils.py:24]: 👍🏼

**Review by goanpeca — APPROVED** [review]
This is great @Zac-HD ! :) 

Minor typo fix

**Review by sofroniewn — APPROVED** [review]
Very nice! I'm approving this. There has already been a lot of enthusiasm for hypothesis on #2444, but I will wait to merge this until tomorrow in case @jni @tlambert03 or anyone else wants to weigh in with additional thoughts before we add it to the repo and our tests

**Review by tlambert03 — APPROVED** [review]
lovely.  looking forward to seeing more examples.  thanks

**Pull request merged**